emily/handler/src/database/accessors.rs

```diff
@@ -242,6 +209,174 @@ async fn wipe_chainstate_table(context: &EmilyContext) -> Result<(), Error> {
     .await
 }
 
+// Chainstate ------------------------------------------------------------------
+
+/// Add a chainstate entry.
+pub async fn add_chainstate_entry(
+    context: &EmilyContext,
+    entry: &ChainstateEntry,
+) -> Result<(), Error> {
+    // Get the existing chainstate entry for height. If there's a conflict
+    // then propagate it back to the caller.
+    let current_chainstate_entry_result = get_chainstate_entry_at_height(context, entry.key.height)
+        .await
+        .and_then(|existing_entry| {
+            if existing_entry != *entry {
```

Comment:
Why would this be wrong? In this moment the entry is a reference to a entry struct, and the `existing_entry` is a struct instance. Am I wrong in thinking that `&existing_entry != entry` would compare the pointer values to the structs and give us the wrong comparison?

Reply:
If this is like `C`, then `&existing_entry` is the numeric pointer to the stack and `entry` is the numeric pointer to the stack from an earlier period.

Reply:
> Why would this be wrong? In this moment the entry is a reference to a entry struct, and the `existing_entry` is a struct instance. 

The reason why this looks wrong to me is because it looks like you're dereferencing the `entry`, but that shouldn't work because the type doesn't implement `Copy` (and/or `Deref`). If you try to break this out as
```rust
let intermediate = *entry;
if existing_entry != intermediate {
...
```
you'll probably get a compiler error. So I was thinking that maybe this is okay because of how `PartialEq` or `Eq` work under the hood, or maybe because the compiler does some magic to make the seemingly equivalent construction work.

> Am I wrong in thinking that `&existing_entry != entry` would compare the pointer values to the structs and give us the wrong comparison?

Yeah, in Rust references are not like pointers, so the reference comparison will actually do what we want.

Reply:
Got it, fixed!